src/matrix/lkj.jl

```diff
@@ -188,32 +189,37 @@ end
 
 function lkj_onion_loginvconst(d::Integer, η::Real)
     #  Equation (17) in LKJ (2009 JMA)
-    sumlogs = zero(η)
-    for k in 2:d - 1
-        sumlogs += 0.5k*logπ + loggamma(η + 0.5(d - 1 - k))
+    T = float(Base.promote_typeof(d, η))
+    h = T(1//2)
+    α = η + h * d - 1
+    loginvconst = (2*η + d - 3)*T(logtwo) + (T(logπ) / 4) * (d * (d - 1) - 2) + logbeta(α, α) - (d - 2) * loggamma(η + h * (d - 1))
+    for k in 2:(d - 1)
+        loginvconst += loggamma(η + h * (d - 1 - k))
     end
-    α = η + 0.5d - 1
-    loginvconst = (2η + d - 3)*logtwo + logbeta(α, α) + sumlogs - (d - 2) * loggamma(η + 0.5(d - 1))
     return loginvconst
 end
 
-function lkj_onion_loginvconst_uniform_odd(d::Integer)
+function lkj_onion_loginvconst_uniform_odd(d::Integer, ::Type{T} = Float64) where {T}
```

Comment:
Do we really want a default value here? The function is neither exported nor part of the API, so we don't need one for backward compatibility.
```suggestion
function lkj_onion_loginvconst_uniform_odd(d::Integer, ::Type{T}) where {T<:Real}
```

Reply:
Done. Thanks.